command/agent/command.go

```diff
@@ -201,9 +202,71 @@ func (c *Command) readConfig() *Config {
 	config.Version = c.Version
 	config.VersionPrerelease = c.VersionPrerelease
 
-	if dev {
-		// Skip validation for dev mode
-		return config
+	// Normalize addresses
+	if config.Addresses.HTTP == "" {
+		config.Addresses.HTTP = fmt.Sprintf("%s:%d", config.BindAddr, config.Ports.HTTP)
+	}
+	if config.Addresses.RPC == "" {
+		config.Addresses.RPC = fmt.Sprintf("%s:%d", config.BindAddr, config.Ports.RPC)
+	}
+	if config.Addresses.Serf == "" {
+		config.Addresses.Serf = fmt.Sprintf("%s:%d", config.BindAddr, config.Ports.Serf)
+	}
+
+	// Use getAdvertise(&config.AdvertiseAddrs.<Service>, <Port>) to
+	// retrieve a default address for advertising if one isn't set.
+	defaultAdvertise := ""
+	getAdvertise := func(addr *string, port int) bool {
+		if *addr != "" {
```

Comment:
Should still be an error if they explicitly advertise `0.0.0.0` or `localhost` (in any of its ipv4 or v6 forms, or via resolution)